mitodl/ol-infrastructure
# Upgrade ocw go to 1.20.4 and hugo to 0.112.0

## Description
This PR updates Go and Hugo versions for mitodl/ocw-course-publisher.

The following are the release notes:
- [Go 1.20.4](https://go.dev/doc/devel/release#go1.20)
- [Hugo 0.112.0](https://github.com/gohugoio/hugo/releases/tag/v0.112.0)

Also fixes `hello.py`.

## Motivation and Context
Closes https://github.com/mitodl/ol-infrastructure/issues/1513

Fixes `hello.py` because why not?

## How Has This Been Tested?
1. `cd mitodl/ol-infrastructure/dockerfiles/ocw/node-hugo`
2. Build the image with `docker build . -t test-hugo`
3. Run the image with `docker run -it test-hugo bash`
4. Then verify the versions with
    ```bash
    hugo version
    /usr/local/go/bin/go version
    ```

## Types of changes

- [x] Enhancement (improves on existing behavior)

## Checklist:


- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
